### PR TITLE
Schema Test Case

### PR DESCRIPTION
**Pull request by justinjoyn**

### CODE REVIEW

File: test/graphql_generator_test.dart [hunk 1]
```diff
@@ -42,4 +42,57 @@ void main() {
       resultFile.delete();
     }
   });
+
+test('GraphQL generation', () async {
+    /// Read the test response from the file.
+    var graphQLGenerator = GraphQLGenerators();
+    var response = await graphQLGenerator.getSchema("https://turtle-api.herokuapp.com/graphiql", "", introspectionQuery:
+    '{"query" : "fragment FullType on __Type { kind name description fields(includeDeprecated: true) '
```
Comment (venkatd) [inline test/graphql_generator_test.dart]: Search for multiline strings in dart. They will make this query more readable.

File: test/graphql_generator_test.dart [hunk 2]
```diff
@@ -42,4 +42,57 @@ void main() {
       resultFile.delete();
     }
   });
+
+test('GraphQL generation', () async {
+    /// Read the test response from the file.
+    var graphQLGenerator = GraphQLGenerators();
+    var response = await graphQLGenerator.getSchema("https://turtle-api.herokuapp.com/graphiql", "", introspectionQuery:
```
Comment (venkatd) [inline test/graphql_generator_test.dart]: Let us avoid calling live APIs. What if the schema changes? See my comment in Turtle about generating your own test introspection schema.

File: test/graphql_generator_test.dart [hunk 3]
```diff
@@ -42,4 +42,57 @@ void main() {
...
+              'query IntrospectionQuery { __schema { queryType { name } mutationType { name } types { ...FullType } '
+              'directives { name description locations args { ...InputValue } } } } "}');
+    var typesFromResponse = getTypeList(
+        graphQLGenerator.getTypesFromResponse(response:response));
+    var mutationTypeName = graphQLGenerator.getMutationTypeNameFromResponse(
+        response: response);
+    var namespace = "T";
+
+    /// Generate the dart code and write it to a file.
+    var resultFile = await File('test/test_data/turtle_backend.schema.result.txt')
```
Comment (venkatd) [inline test/graphql_generator_test.dart]: Why are you writing it to a string? It seems like an unnecessary step to write the result from a file, pull it back out of the file, then delete the file when you are done. I would suggest storing the result in a string and skipping all fo this.

**Comment by venkatd** [comment]
Thanks for the PR on this. I left some suggestions.

File: test/graphql_generator_test.dart (hunk 3 above)
Comment (justinjoyn) [reply test/graphql_generator_test.dart]: There was an issue with the '\n' (newline) when the result string was compared to the string from file. 
When I save the result to the file and retrieve it as a string, the problem was resolved.

File: test/graphql_generator_test.dart (hunk 2 above)
Comment (justinjoyn) [reply test/graphql_generator_test.dart]: @venkatd If we are not using a live API we would have to pass the response JSON directly to the library. Is that how we need to do this?

**Comment by venkatd** [comment]
Ok let us merge this for now and can improve later

**Pull request merged**